code/modules/power/tesla/energy_ball.dm

```diff
@@ -196,12 +199,30 @@
 
 
 /obj/singularity/energy_ball/Collide(atom/A)
-	. = ..()
-	dust_mobs(A)
+	if(isliving(A))
+		dust_mobs(A)
+	else if(isobj(A))
+		var/obj/O = A
+		O.tesla_act(0, TRUE)
 
 /obj/singularity/energy_ball/CollidedWith(atom/A)
+	if(isliving(A))
+		dust_mobs(A)
+	else if(isobj(A))
+		var/obj/O = A
+		O.tesla_act(0, TRUE)
+
+/obj/singularity/energy_ball/Move(NewLoc, Dir)
 	. = ..()
-	dust_mobs(A)
+	for(var/v in view(0, loc))
```

Comment:
Why not just: `for (var/v in loc)`

Reply:
Also, why is this necessary; shouldn't collidedwith take care of this logic?

Reply:
Because I want to ignore things under Turf. It eats all pipes/cables which is not what we want.